go-runtime/compile/schema.go

```diff
@@ -385,16 +386,60 @@ func goNodePosToSchemaPos(node ast.Node) (schema.Position, int) {
 	return schema.Position{Filename: p.Filename, Line: p.Line, Column: p.Column, Offset: p.Offset}, fset.Position(node.End()).Column
 }
 
+func maybeVisitTypeEnumVariant(pctx *parseContext, node *ast.GenDecl, directives []directive) bool {
+	if len(node.Specs) != 1 {
+		return false
+	}
+	// If any directives on this node are exported, then the node is considered exported for type enum variant purposes
+	isExported := false
+	for _, dir := range directives {
```

Comment:
the enum variants won't have their own enum/export directive so i think this will always return false—`//ftl:enum` annotates the outer enum type rather than the variants. so i think we need to pull this data from the enum at `pctx.enums[enumName]` (and make sure it's set when we create this entry), e.g.
`isExported := pctx.enums[enumName].Export`. passing this in to `visitType` will handle exporting any variant types we need to transitively

let's also make sure this is tested

Reply:
It's true if one of the variants is already an exported struct (tested in two.go L#30). Basically, if it's already exported for some other reason, and then someone decides to add it as a variant to a type enum as well, then we end up with this, so we have to pass through that directive's own `.Export` so that the `visitType` call in the helper function works correctly.

Reply:
I spent some more time thinking about this and we'll have to look at both. The enum variants (e.g. the `Exported` struct in the test) can separately be exported as well, even if the enum itself is not exported. This `isExported` here needs to be true if _either_ the parent enum _or_ the variant itself is exported. I'll update the code to include that and add some tests.

Reply:
good point! let's give priority to the explicit annotation—so if present it overrides the visibility that comes in transitively. and in validation we should check that all external enum refs are actually exported

Reply:
done :D